File: expandablrecyclerview/src/main/java/com/umut/expandablrecyclerview/adapter/holder/ChildViewHolder.java
package com.umut.expandablrecyclerview.adapter.holder;

import androidx.annotation.NonNull;
import android.view.View;

import com.umut.expandablrecyclerview.adapter.ChildCoordinate;

import java.util.List;

public abstract class ChildViewHolder extends ExpandableViewHolder<ChildCoordinate> {

    public ChildViewHolder(View itemView) {
        super(itemView);
    }

    /**
     * binds the value to view holder
     *
     * @param coordinate Coordinate of item
     */
    public abstract void bind(@NonNull ChildCoordinate coordinate);

    public boolean update(@NonNull List<Object> payload, ChildCoordinate coordinate) {
        return false;
    }
}

File: expandablrecyclerview/src/main/java/com/umut/expandablrecyclerview/adapter/holder/ChildViewHolder.kt
package com.umut.expandablrecyclerview.adapter.holder

import android.view.View

import com.umut.expandablrecyclerview.adapter.ChildCoordinate

abstract class ChildViewHolder(itemView: View) : ExpandableViewHolder<ChildCoordinate>(itemView) {

    /**
     * binds the value to view holder
     *
     * @param coordinate Coordinate of item
     */
    abstract fun bind(coordinate: ChildCoordinate)

    fun update(payload: List<Any>, coordinate: ChildCoordinate): Boolean {
        return false
    }
}
